live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/filter/RepoEventFilterHandler.java

```diff
@@ -42,13 +49,61 @@
 @Slf4j
 public class RepoEventFilterHandler
 {
-
     private final List<RepoEventFilterApplier> repoEventFilterAppliers;
+    private final CamelEventMapper camelEventMapper;
+
+    /**
+     * Method handles event/node filtering. When node is denied by any of filters, a custom property "DENY_NODE: true" in Camel Exchange is created (used to filter out in main route). In case of update event: - when current version of the node is allowed but previous version was denied, then the type of the event is altered to 'Created'. - when current version of the node is denied but previous version was allowed, then the type of the event is altered to 'Deleted'.
+     *
+     * @param exchange
+     *            Camel Exchange Object
+     * @param filter
+     *            Filter configuration
+     */
+    public boolean handleAndGetAllowed(Exchange exchange, Filter filter)
+    {
+        final FilteringResults filteringResults = calculateFilteringResults(exchange, filter);
+        filteringResults.newEventType.ifPresent(type -> exchange.getIn().setBody(camelEventMapper.alterRepoEvent(exchange, type)));
+        return filteringResults.allowed;
+    }
 
-    public boolean filterNode(RepoEvent<DataAttributes<NodeResource>> repoEvent, Filter filter)
+    private FilteringResults calculateFilteringResults(Exchange exchange, Filter filter)
     {
-        return repoEventFilterAppliers.stream()
-                .peek(f -> log.atDebug().log("Applying filters {} to repo event of id: {}", filter, repoEvent.getId()))
-                .allMatch(f -> f.applyFilter(repoEvent, filter));
+        final RepoEvent<DataAttributes<NodeResource>> repoEvent = exchange.getIn().getBody(RepoEvent.class);
+        boolean allowCurrentNode = true;
+        boolean allowPreviousNode = true;
+        final boolean eventTypeUpdated = EventUtils.isEventTypeUpdated(repoEvent);
+        for (RepoEventFilterApplier filterApplier : repoEventFilterAppliers)
+        {
+            log.atDebug().log("Applying filters {} to current repo event of id: {}", filter, repoEvent.getId());
+            final boolean allow = filterApplier.allowNode(repoEvent.getData().getResource(), filter);
+            allowCurrentNode = allowCurrentNode && allow;
+            if (eventTypeUpdated)
```

Comment:
Really minor performance improvement:
```suggestion
            if (eventTypeUpdated && allowPreviousNode)
```

Reply:
Applied. Also got rid of one unnecessary (after this change) local variable below.